nested/diskSelection.json

```diff
@@ -0,0 +1,92 @@
+    "outputs": {
+        "dataDiskArray": {
+            "type": "array",
+            "value": "[take(variables('diskArray'),parameters('vmDiskCount'))]"
+        }
+    },
+    "parameters": {
+        "vmDiskCount": {
+            "maxValue": 8,
+            "minValue": 2,
+            "type": "int"
+        },
+        "vmName": {
+            "type": "string"
+        }
+    },
+    "resources": [],
+    "variables": {
+        "diskArray": [
+            {
+                "caching": "None",
+                "createOption": "Attach",
+                "lun": 0,
+                "managedDisk": {
+                    "id": "[resourceId('Microsoft.Compute/disks', concat(parameters('vmName'),'-datadisk1'))]"
+                }
+            },
+            {
+                "caching": "None",
+                "createOption": "Attach",
+                "lun": 1,
+                "managedDisk": {
+                    "id": "[resourceId('Microsoft.Compute/disks', concat(parameters('vmName'),'-datadisk2'))]"
+                }
+            },
+            {
+                "caching": "None",
+                "createOption": "Attach",
+                "lun": 2,
+                "managedDisk": {
+                    "id": "[resourceId('Microsoft.Compute/disks', concat(parameters('vmName'),'-datadisk3'))]"
+                }
+            },
+            {
+                "caching": "None",
+                "createOption": "Attach",
+                "lun": 3,
+                "managedDisk": {
+                    "id": "[resourceId('Microsoft.Compute/disks', concat(parameters('vmName'),'-datadisk4'))]"
+                }
+            },
+            {
+                "caching": "None",
+                "createOption": "Attach",
+                "lun": 4,
+                "managedDisk": {
+                    "id": "[resourceId('Microsoft.Compute/disks', concat(parameters('vmName'),'-datadisk5'))]"
+                }
+            },
+            {
+                "caching": "None",
+                "createOption": "Attach",
+                "lun": 5,
+                "managedDisk": {
+                    "id": "[resourceId('Microsoft.Compute/disks', concat(parameters('vmName'),'-datadisk6'))]"
+                }
+            },
+            {
+                "caching": "None",
+                "createOption": "Attach",
+                "lun": 6,
+                "managedDisk": {
+                    "id": "[resourceId('Microsoft.Compute/disks', concat(parameters('vmName'),'-datadisk7'))]"
+                }
+            },
+            {
+                "caching": "None",
+                "createOption": "Attach",
+                "lun": 7,
+                "managedDisk": {
+                    "id": "[resourceId('Microsoft.Compute/disks', concat(parameters('vmName'),'-datadisk8'))]"
+                }
+            }
+        ],
+        "documentation01": "This sub-template provides a way to return a flexible number of drive configurations",
```

Comment:
use a property copy loop for this instead of the nested deployments, see: https://github.com/Azure/azure-quickstart-templates/blob/master/101-vm-windows-copy-datadisks/azuredeploy.json#L150-L160

Reply:
This is great. Modified as such and removed this nested template.

Reply:
I was excited to know about this and try it, but unfortunately, it doesn't work with our current template structure. ARM doesn't allow 0 copy count, but we are depending on an empty dataDisks array, so a property copy loop can't be used in that case (in the controller.json). So I had to revive this nested template, after so many trials and errors. I wish ARM supported 0 copy count...

Reply:
Giving the take() a try, which seems to accept 0 as the count. Quite inconsistent...